From 0e12ee904bbbb9e3bcfd3b2e0c9d4d2d715166f0 Mon Sep 17 00:00:00 2001
From: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Date: Fri, 5 Apr 2019 06:15:52 -0700
Subject: [PATCH] dlopen() soname instead of linker name

Fixes: #60

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f1c7634..43b9ec4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,7 +144,7 @@ add_definitions( -D__STDC_LIMIT_MACROS )
 add_definitions( -D__STDC_CONSTANT_MACROS )
 add_definitions( -DCOMMON_CLANG_EXPORTS )
 if (NOT WIN32)
-    add_definitions( -DLIBCOMMON_CLANG_NAME="lib${COMMON_CLANG_LIBRARY_NAME}.so")
+    add_definitions(-DLIBCOMMON_CLANG_NAME="lib${COMMON_CLANG_LIBRARY_NAME}.so.${LLVM_VERSION_MAJOR}")
 endif()
 
 #
